src/test/java/com/epam/tc/hw5/steps/UserTableSteps.java

```diff
@@ -0,0 +1,109 @@
+package com.epam.tc.hw5.steps;
+
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class UserTableSteps extends AbstractStep {
+
+    @Then("\"User Table\" page should be opened")
+    public void userTableIsOpened() {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(By.className("uui-main-container")));
+        softly.assertThat(driver.getCurrentUrl())
+                .as("Incorrect page opened")
+                .isEqualTo(userTablePage.getUserTablePageURL());
+        softly.assertAll();
+    }
+
+    @Then("{string} Number Type Dropdowns should be displayed on Users Table on User Table Page")
+    public void numberOfTypeDropdownsIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getNumberTypeDropdownsSize())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("{string} Usernames should be displayed on Users Table on User Table Page")
+    public void numberOfUsernamesIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getUserNamesNumber())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("{string} Description texts under images should be displayed on Users Table on User Table Page")
+    public void numberOfDescriptionTextsIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getDescriptionTextsNumber())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("{string} checkboxes should be displayed on Users Table on User Table Page")
+    public void numberOfCheckboxesIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getNumberOfCheckboxes())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("User table should contain following values:")
+    public void user_table_should_contain_following_values(io.cucumber.datatable.DataTable dataTable) {
+        List<List<String>> userTable = dataTable.cells().subList(1, 7);
```

Comment:
Are you sure about magic numbers 1 and 7? Why 7? If the table contains 8 values will your method work correctly?

Reply:
Fixed